TypeCobol/Compiler/Parser/CodeElementBuilder/TypeCobolCodeElementBuilder.cs

```diff
@@ -1,208 +1,421 @@
+                    }
+
+                    //Add outputs to global callsites parameters
+                    if (outputs.Count > 0)
+                    {
+                        foreach (var param in outputs)
+                        {
+                            callSite.Parameters[i] = param;
+                            i++;
+                        }
+                    }
+
+                    //Add inouts to global callsites parameters
+                    if (inouts.Count > 0)
+                    {
+                        foreach (var param in inouts)
+                        {
+                            callSite.Parameters[i] = param;
+                            i++;
+                        }
+                    }
+
+                    #endregion
+                }
+                else //else it's an error
+                    isInError = true;
+            }
+            else
+            {
+                if (ambiguousSymbol != null &&
+                    (ambiguousSymbol.Type == SymbolType.ProgramEntry || ambiguousSymbol.Type == SymbolType.ProgramName ||
+                     ambiguousSymbol.CandidateTypes.Any(c => c == SymbolType.ProgramEntry || c == SymbolType.ProgramName)))
+                {
+                    //if CandidatesTypes = Program or program Entry
+                    //new ProcedureStyleCallStatement() { ProgramNameOrProgramEntry= temp}
+                    ((AmbiguousSymbolReference) ambiguousSymbolReference.MainSymbolReference).CandidateTypes =
+                       ambiguousSymbol.CandidateTypes.Where(
+                           c => c == SymbolType.ProgramEntry || c == SymbolType.ProgramName).ToArray();
+                    statement =
+                        new ProcedureStyleCallStatement(new ProcedureCall(ambiguousSymbolReference.MainSymbolReference,
+                            null, null, null))
+                        {
+                            ProgramNameOrProgramEntry = ambiguousSymbolReference.MainSymbolReference
+                        };
+                    callSite.CallTarget = statement.ProgramNameOrProgramEntry;
+                }
+                else if (ambiguousSymbol != null &&
+                         (ambiguousSymbol.Type == SymbolType.DataName ||
+                          ambiguousSymbol.Type == SymbolType.TCFunctionName ||
+                          ambiguousSymbol.CandidateTypes.Any(
+                              c => c == SymbolType.DataName || c == SymbolType.TCFunctionName)))
+                {
+                    //else if CandidatesTypes = data or TCFunctionName
+                    //new ProcedureStyleCallStatement() {ProcdurePointerOrTCProcedureFunction = temp}
+                    ((AmbiguousSymbolReference) ambiguousSymbolReference.MainSymbolReference).CandidateTypes =
+                        ambiguousSymbol.CandidateTypes.Where(
+                            c => c == SymbolType.DataName || c == SymbolType.TCFunctionName).ToArray();
```

Comment:
You can directly create an array with SymbolType.DataName and SymbolType.TCFunctionName